src/_cffi_src/openssl/x509.py

```diff
@@ -350,6 +354,33 @@
 """
 
 CUSTOMIZATIONS = """
+/* Added in 1.0.2 beta but we need it in all versions now due to the great
+   opaquing. */
+#if OPENSSL_VERSION_NUMBER < 0x10002001L || defined(LIBRESSL_VERSION_NUMBER)
+/* from x509/x_x509.c version 1.0.2 */
+void X509_get0_signature(ASN1_BIT_STRING **psig, X509_ALGOR **palg,
+                         const X509 *x)
+{
+    if (psig)
+        *psig = x->signature;
+    if (palg)
+        *palg = x->sig_alg;
+}
+#endif
+/* Added in 1.0.2 but we need it in all versions now due to the great
+   opaquing. */
+#if OPENSSL_VERSION_NUMBER < 0x10002003L || defined(LIBRESSL_VERSION_NUMBER)
+/* from x509/x_x509.c */
+int i2d_re_X509_tbs(X509 *x, unsigned char **pp)
+{
+    /* In 1.0.1 and below cert_info is a pointer in the struct, so
+       we don't want to pass by reference. */
```

Comment:
That's because it makes little sense. I just tried to figure out what I meant. I *think* I was trying to say that in 1.1.0 the opaque X509 struct makes cert_info a direct member of the struct rather than a pointer. So in 1.1.0 the function definition looks like `return i2d_X509_CINF(&x->cert_info, pp);`. But this isn't a particularly useful comment so I should just remove it.